### PR TITLE
Move email config in files

### PR DESCRIPTION
**Pull request by petersg83**

### CODE REVIEW

File: docs/3.0.0-beta.x/plugins/email.md [hunk 1]
```diff
@@ -54,28 +50,43 @@ npm install strapi-provider-email-sendgrid@beta --save
 
 ::::
 
+**Note:** If the provider is not in the mono repo, you probably don't need `@beta` depending if the creator published it with this tag or not.
```
Comment (alexandrebodin) [inline docs/3.0.0-beta.x/plugins/email.md:53]: We can remove this note as we already removed the need for @beta

File: packages/strapi-admin/controllers/Auth.js [hunk 2]
```diff
@@ -335,12 +330,7 @@ module.exports = {
       // Send an email to the admin.
       await strapi.plugins['email'].services.email.send({
         to: admin.email,
-        from:
-          settings.from.email || settings.from.name
-            ? `${settings.from.name} <${settings.from.email}>`
-            : undefined,
-        replyTo: settings.response_email,
-        subject: settings.object,
+        subject: 'Reset password',
```
Comment (alexandrebodin) [inline packages/strapi-admin/controllers/Auth.js:333]: I don't understand where all the other params went ? :D

File: packages/strapi-provider-email-mailgun/lib/index.js [hunk 3]
```diff
@@ -11,35 +11,11 @@ const mailgunFactory = require('mailgun-js');
...
-      label: 'Mailgun Domain',
-      type: 'text',
-    },
-  },
   init: config => {
     const mailgun = mailgunFactory({
-      apiKey: config.mailgun_api_key,
-      host: config.mailgun_api_host,
-      domain: config.mailgun_domain,
+      apiKey: config.apiKey,
```
Comment (alexandrebodin) [inline packages/strapi-provider-email-mailgun/lib/index.js:16]: One thing we could do is to send the full config object instead so the user can use the full extend of the provider options and in the readme we can document what provider is used to know more about the other params. What do you think ?

File: packages/strapi-admin/controllers/Auth.js (hunk 2 above)
Comment (petersg83) [reply packages/strapi-admin/controllers/Auth.js:333]: They went in the cloud :p

I deleted the hard coded  `from` address to reverse the code introduced by this PR : https://github.com/strapi/strapi/pull/5584

File: packages/strapi-provider-email-sendgrid/lib/index.js [hunk 4]
```diff
@@ -1,58 +1,31 @@
...
+            cc,
+            bcc,
+            replyTo: replyTo || settings.defaultReplyTo,
+            subject,
+            text,
+            html,
+            ...rest,
           };
 
+          msg = _.pickBy(msg, value => typeof value !== 'undefined');
```
Comment (alexandrebodin) [inline packages/strapi-provider-email-sendgrid/lib/index.js:27]: You can create a function removeUndefined and call sendgrid.send(removeUndefined(msg))  readability's sake

File: docs/3.0.0-beta.x/plugins/email.md [hunk 5]
```diff
@@ -11,6 +11,8 @@ By using the following function, strapi will use the configured provider to send
 await strapi.plugins['email'].services.email.send({
   to: 'paulbocuse@strapi.io',
   from: 'joelrobuchon@strapi.io',
+  cc: 'lauralumale@strapi.io',
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/plugins/email.md:14]: Can you please change these emails.
We should use Chefs names.

File: docs/3.0.0-beta.x/plugins/email.md [hunk 6]
```diff
@@ -11,6 +11,8 @@ By using the following function, strapi will use the configured provider to send
...
+  bcc: 'arthurpapailhau@strapi.io',
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/plugins/email.md:15]: Same here.

File: docs/3.0.0-beta.x/plugins/email.md [hunk 7]
```diff
@@ -54,29 +50,44 @@ npm install strapi-provider-email-sendgrid@beta --save
...
+```js
+module.exports = ({ env }) => ({
+  // ...
+  email: {
+    provider: 'sendgrid',
+    providerOptions: {
+      apiKey: env('SENDGRID_API_KEY'),
+    },
+    settings: {
+      defaultFrom: 'myemail@protonmail.com',
```
Comment (lauriejim) [inline docs/3.0.0-beta.x/plugins/email.md:70]: Same here.

**Pull request merged**